malariagen_data/util.py

```diff
@@ -918,7 +917,42 @@ def plotly_discrete_legend(
     return fig
 
 
-def check_colab_location(*, gcs_url: str, url: str) -> Optional[ipinfo.details.Details]:
+def get_gcp_region(details):
+    """Attempt to determine the current GCP region based on
+    response from ipinfo."""
+
+    if details is not None:
+        org = details.org
+        country = details.country
+        region = details.region
+        if org == "AS396982 Google LLC":
+            if country == "US":
+                if region == "Iowa":
+                    return "us-central1"
+                elif region == "South Carolina":
+                    return "us-east1"
+                elif region == "Virginia":
+                    return "us-east4"
+                elif region == "Ohio":
+                    return "us-east5"
+                elif region == "Oregon":
+                    return "us-west1"
+                elif region == "California":
+                    return "us-west2"
+                elif region == "Utah":
+                    return "us-west3"
+                elif region == "Nevada":
+                    return "us-west4"
+                elif region == "Texas":
+                    return "us-south1"
+            elif country == "ZA":
+                if region == "Gauteng":
+                    return "africa-south1"
```

Comment:
@alimanfoo More of a question. Is there a benefit on listing regions directly here now? Is this thinking ahead on a scenario where the number of resources/regions increases in Africa, then more of the VMs get allocated in `africa-south1` & then, we might consider a new single region bucket for this region?

Reply:
Yeah vaguely, I was just going to do the US regions for now but then happened to notice that the African region is up, and so added it here because I had the mapping from ipinfo region ("Gauteng") to GCP region ("africa-south1") to hand. 

But yes, at some point it would be cool to add a bucket in an African region.